.github/workflows/auto-cherry-picker.yaml

```diff
@@ -0,0 +1,47 @@
+name: Auto Cherry-Picker
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      PR_number:
+        description: The PR number to cherry-pick
+        type: string
+        required: true
+
+jobs:
+  inputs:
```

Comment:
Would it be possible to have the two jobs here run as a single job, for lower latency and simpler workflow? I think the `checkout` step can refer to an earlier step's output similar to a job's output: https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-steps-context

Reply:
Nothing is simple about GitHub actions.

IMO this was the simpler approach.

Reply:
I think the `step1` step can be plopped into the `picker` job, and the `${{ needs... }}` reference changed to `${{ steps.step1.outputs.merge_commit }}`, and everything else can disappear, which makes this seem strictly simpler (I know I always make mistakes with/find it harder to understand anything that goes across jobs, personally).


```yaml
jobs:
  picker:
    if: ... # as it is now
    runs-on: ubuntu-latest
    steps:
      - id: step1
        ... # as it is now

      - name: Check out code
        uses: actions/checkout@v2
        with:
          ref: ${{ steps.step1.outputs.merge_commit }} # adjusted

      ... # as it is now
```

Reply:
I actually had it like this but ended up just putting it in one job :|

Reply:
Let's do the other comments and then we can swing back around.